client/src/components/TodoListItem.tsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import '../App.css';
+import {graphql} from 'babel-plugin-relay/macro';
+import { useFragment, commitMutation } from 'react-relay/hooks';
+import type {TodoListItem_todo$key} from './__generated__/TodoListItem_todo.graphql';
+import Checkbox from '@button-inc/component-library/Checkbox'
+import RelayEnvironment from '../RelayEnvironment';
+
+type Props = {
+    //$key includes data and fragment reference
+    todo: TodoListItem_todo$key
+}
+
+const UpdateTodoMutation = graphql`
+mutation TodoListItemUpdateMutation (
+  $input: UpdateTodoInput!) {
+  updateTodo(input: $input) {
+    todoEdge {
+      node {
+        completed
```

Comment:
you'll need to fetch the id here, for Relay to update its local store
https://relay.dev/docs/guided-tour/updating-data/graphql-mutations/#updating-data-once-a-request-is-complete

This way if the request fails, the checkbox will uncheck itself on re-render